HunterPie.Integrations/Datasources/MonsterHunterRise/Entity/Player/MHRPlayer.cs

```diff
@@ -416,10 +416,18 @@ private void GetConsumableAbnormalities()
                 abnormality.Timer = isConditionValid ? 1 : 0;
             else if (isConditionValid)
             {
-                abnormality = Process.Memory.Read<MHRConsumableStructure>(consumableBuffs + schema.Offset);
+                if (schema.IsInteger)
+                    abnormality.Timer = Process.Memory.Read<int>(consumableBuffs + schema.Offset);
+                else
+                {
+                    abnormality = Process.Memory.Read<MHRConsumableStructure>(consumableBuffs + schema.Offset);
```

Comment:
Instead of having more nested if/elses to read different types, you can just create a union and then access the value with the right type:

```cs
[StructLayout(LayoutKind.Explicit)]
public struct MHRAbnormalityStructure
{
   [FieldOffset(0)] public int Timer;
   [FieldOffset(0)] public MHRConsumableStructure Consumable;
   [FieldOffset(0)] public MHRDebuffStructure Debuff;
}

```

And then create an adapter that receives a `MHRAbnormalityStructure` and a schema and returns some kind of `MHRAbnormalityData` with the right timer

Reply:
My idea is to create another structure like
```cs
public struct MHRAbnormalityData
{
    public float Timer;
}
```
And an adapter:
```cs
public static class MHRAbnormalityAdapter
{
    public static MHRAbnormalityData Convert(AbnormalitySchema schema, MHRAbnormalityStructure abnormality) 
    {
        if (schema.IsInteger)
            return new MHRAbnormalityData { Timer: (float)abnormality.Timer };

        return schema.Group switch 
        {
            "Debuffs" => return new MHRAbnormalityData { Timer: abnormality.Debuff.Timer },
            "Consumables" =>  return new MHRAbnormalityData { Timer: abnormality.Consumable.Timer },
            _ => new MHRAbnormalityData() 
        }
    }
}
```

Then you can use that in the `MHRPlayer`

```cs
[...]
else if (isConditionValid) 
{
    var abnormalityStructure = Memory.Read<MHRAbnormalityStructure>(ptr + schema.Offset);
    var abnormality = MHRAbnormalityAdapter.Convert(schema, abnormalityStructure);

    // Do the rest of code here
}

HandleAbnormality<MHRConsumableAbnormality, MHRAbnormalityData>(
    // ...params
)
```

Reply:
Works well.